js/app.js

```diff
@@ -0,0 +1,252 @@
+// 1
+const user = {
+    firstName: 'John',
+    lastName: 'Smith',
+    age: undefined
+};
+
+function checkIfPropertyIsDefined(user, property) {
+    if (user.hasOwnProperty(property) && user[property] !== undefined) {
+        return true;
+    }
+    return false;
+}
+
+console.log(checkIfPropertyIsDefined(user, 'firstName')); // true
+console.log(checkIfPropertyIsDefined(user, 'age')); // false
+console.log(checkIfPropertyIsDefined(user, 'height')); // false
+
+function checkIfPropertyIsDefined(user, property) {
+    return user.hasOwnProperty(property) && user[property] !== undefined;
+}
+
+console.log(checkIfPropertyIsDefined(user, 'firstName')); // true
+console.log(checkIfPropertyIsDefined(user, 'age')); // false
+console.log(checkIfPropertyIsDefined(user, 'height')); // false
+
+// 2
+const salaries = {
+    john: 50,
+    adam: 100,
+    katie: 75
+}
+
+const salarySum = salaries.john + salaries.adam + salaries.katie;
+
+console.log(salarySum);
+
+// 3
+console.log(typeof {}); // console logs object so I've to use "object"
+console.log(typeof null); // I'didn't expect that null is also an object
+
+function isObject(value) {
+    return typeof value === 'object' && value !== null;
+}
+
+console.log(isObject({})); // true
+console.log(isObject(10)); // false
+console.log(isObject(null)); // false
+
+// 4
+function removeProperty(user4, key) {
```

Comment:
```suggestion
function removeProperty(user, key) {
```

Reply:
Here you are declaring an argument which will be available only in the scope of the removeProperty function. No reason to name it `user4`